tools/lint/changed_lines.py

```diff
@@ -0,0 +1,170 @@
+    return git_request.json()
+
+
+def parse_patch_file(entry):
+    """Parses the individual file changes within a patch"""
+    line_array = []
+    sublist = []
+
+    patch_array = re.split("\n", entry["patch"])
+    # clean patch array
+    patch_array = [i for i in patch_array if i]
+
+    for item in patch_array:
+        # Grabs hunk annotation and strips out added lines
+        if item.startswith("@@ -"):
+            if sublist:
+                line_array.append(sublist)
+            sublist = [re.sub(r"\s@@(.*)", "", item.split("+")[1])]
+        # We don't need removed lines ('-')
+        elif not item.startswith("-") and not item == "\\ No newline at end of file":
+            sublist.append(item)
+    if sublist:
+        line_array.append(sublist)
+    return line_array
+
+
+def parse_patch_data(patch_data):
+    """Takes the patch data and returns a dictionary of files and the lines"""
+
+    final_dict = {}
+    for entry in patch_data:
+        # We don't need removed files
+        if entry["status"] == "removed":
+            continue
+
+        # We can only operate on files with additions and a patch key
+        # Some really big files don't have a patch key because GitHub
+        # returns a message in the PR that the file is too large to display
+        if entry["additions"] != 0 and "patch" in entry:
+            line_array = parse_patch_file(entry)
+            final_dict[entry["filename"]] = line_array
+    return final_dict
+
+
+def get_lines(line_dict):
+    """Takes the dictionary of files and lines and returns a dictionary of files and line numbers"""
+    final_dict = {}
+    for file_name, sublist in line_dict.items():
+        line_array = []
+        for array in sublist:
+            line_number = 0
+            if "," not in array[0]:
+                line_number = int(array[0]) - 1
+            else:
+                line_number = int(array[0].split(",")[0]) - 1
+
+            start = -1
+            end = -1
+            for line in array:
+                if line.startswith("+"):
+                    if start < 0:
+                        start = line_number
+                    end = line_number
+                    # line_array.append(line_number)
+                line_number += 1
+            line_array.append([start, end])
+
+        # Remove deleted/renamed files (which appear as empty arrays)
+        if line_array:
+            final_dict[file_name] = line_array
+    return final_dict
+
+
+def main():
+    """main()"""
+    parser = argparse.ArgumentParser(
+        prog="changed_lines.py",
+        description="Identifies the changed files and lines in a GitHub PR.",
+    )
+    parser.add_argument("--token")
+    parser.add_argument("--api-url", default="https://api.github.com")
+    parser.add_argument("--repo", default="FreeCAD/FreeCAD")
+    parser.add_argument("--ref", required=True)
+    parser.add_argument("--pr", required=True)
+    parser.add_argument("--delimeter", default=" ")
```

Comment:
This is misspelled, it should be "delimiter":
```suggestion
    parser.add_argument("--delimiter", default=" ")
```
I don't see any uses of it in the scripts, but I will hold off on merging until @oursland confirms.

Reply:
I have removed it as it is no longer used.